aws/resource_aws_cloudformation_stack_set.go

```diff
@@ -250,45 +250,43 @@ func resourceAwsCloudFormationStackSetUpdate(d *schema.ResourceData, meta interf
 func resourceAwsCloudFormationStackSetDelete(d *schema.ResourceData, meta interface{}) error {
 	conn := meta.(*AWSClient).cfconn
 
-	input := &cloudformation.DeleteStackSetInput{
-		StackSetName: aws.String(d.Id()),
-	}
-
 	log.Printf("[DEBUG] Deleting CloudFormation StackSet: %s", d.Id())
-	_, err := conn.DeleteStackSet(input)
-
-	if isAWSErr(err, cloudformation.ErrCodeStackSetNotFoundException, "") {
-		return nil
-	}
-
+	err := deleteCloudFormationStackSet(conn, d.Id())
 	if err != nil {
 		return fmt.Errorf("error deleting CloudFormation StackSet (%s): %s", d.Id(), err)
 	}
 
 	return nil
 }
 
-func listCloudFormationStackSets(conn *cloudformation.CloudFormation) ([]*cloudformation.StackSetSummary, error) {
+func deleteCloudFormationStackSet(conn *cloudformation.CloudFormation, name string) error {
```

Comment:
Any thoughts around having the resource manipulation functions in a service-specific package (e.g `internal/service/cloudformation/actions`)?

The implementation becomes

```go
func DeleteStackSet(conn *cloudformation.CloudFormation, name string) error {...}
```

and the caller

```go
err := actions.DeleteStackSet(conn, d.Id())
```

It may make code generation easier.

Reply:
Yes, they would definitely belong there

Reply:
It should definitely be moved to something like that. We'll have to figure out where to move `isAWSErr()` first, because otherwise we'll have a dependency loop

Reply:
https://github.com/terraform-providers/terraform-provider-aws/pull/13036 should help with that.

Reply:
Thanks, I was about to create an issue for that